components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -440,7 +441,15 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
 
         } else {
             try {
-                accessTokenDO = OAuth2Util.findAccessToken(validationRequest.getAccessToken().getIdentifier(), false);
+                int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
can we verify whether were we using the tenant id from the thread local in the previous implementations?

Reply:
Good point. There might be components consuming this at an OSGi or service layer. So would be good to validate this

Reply:
Yes I could see we are using this in the framework[1] and oauth[2]

[1] https://github.com/wso2/carbon-identity-framework/blob/b297e3006bb65a159050d2a0861a42235759fbfb/components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/ChallengeQuestionProcessor.java#L62
[2] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/e16d6d7f40982b7d4d021dfc9e0e3ff2358e8835/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java#L127

Reply:
Previously we used the tenant domain from carbon context [1]. 

[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java#L963

Reply:
This PR is created just to move the logic introduced by [1] to the service layer.  Hence created another issue[1] to track the issue with consuming in java class
[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1871
[2] https://github.com/wso2/product-is/issues/14966